### PR TITLE
transport integration tests: add tests for resource manager

### PR DESCRIPTION
**Pull request by MarcoPolo**
- Remove duplicate `SetProtocol` call
- Don't log if transport ErrListenerClosed
- Close the conn scope in webtransport dial
- Mock resource scope span
- Add transport rcmgr integration test

Related to #2194

### CODE REVIEW

File: p2p/test/transport/rcmgr_test.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+)
+
+func TestResourceManagerIsUsed(t *testing.T) {
+	for _, tc := range transportsToTest {
+		t.Run(tc.Name, func(t *testing.T) {
+			for _, testDialer := range []bool{true, false} {
+				t.Run(tc.Name+fmt.Sprintf(" test_dialer=%v", testDialer), func(t *testing.T) {
+
+					reservedMemory := atomic.Uint32{}
+					releasedMemory := atomic.Uint32{}
```
Comment (marten-seemann) [inline p2p/test/transport/rcmgr_test.go:26]:
```suggestion
					var reservedMemory, releasedMemory atomic.Uint32
```

File: p2p/test/transport/rcmgr_test.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/libp2p/go-libp2p/p2p/protocol/ping"
+	"github.com/stretchr/testify/require"
+)
+
+func TestResourceManagerIsUsed(t *testing.T) {
+	for _, tc := range transportsToTest {
+		t.Run(tc.Name, func(t *testing.T) {
+			for _, testDialer := range []bool{true, false} {
+				t.Run(tc.Name+fmt.Sprintf(" test_dialer=%v", testDialer), func(t *testing.T) {
```
Comment (marten-seemann) [inline p2p/test/transport/rcmgr_test.go:23]: Is it ok to capture the loop variable here?

File: p2p/test/transport/rcmgr_test.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+					})
+					peerScope.EXPECT().BeginSpan().AnyTimes().DoAndReturn(func() (network.ResourceScopeSpan, error) {
+						s := mocknetwork.NewMockResourceScopeSpan(ctrl)
+						s.EXPECT().BeginSpan().AnyTimes().Return(mocknetwork.NewMockResourceScopeSpan(ctrl), nil)
+						// No need to track these memory reservations since we assert that Done is called
+						s.EXPECT().ReserveMemory(gomock.Any(), gomock.Any())
+						s.EXPECT().Done()
+						return s, nil
+					})
+					calledSetPeer := atomic.Bool{}
```
Comment (marten-seemann) [inline p2p/test/transport/rcmgr_test.go:74]:
```suggestion
					var calledSetPeer atomic.Bool
```

File: p2p/test/transport/rcmgr_test.go [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+						streamScope.EXPECT().ReleaseMemory(gomock.Any()).AnyTimes()
+						streamScope.EXPECT().BeginSpan().AnyTimes().DoAndReturn(func() (network.ResourceScopeSpan, error) {
+							s := mocknetwork.NewMockResourceScopeSpan(ctrl)
+							s.EXPECT().BeginSpan().AnyTimes().Return(mocknetwork.NewMockResourceScopeSpan(ctrl), nil)
+							s.EXPECT().Done()
+							return s, nil
+						})
+						streamScope.EXPECT().Done()
+
+						streamScope.EXPECT().SetService(gomock.Any()).MinTimes(0).MaxTimes(1)
```
Comment (marten-seemann) [inline p2p/test/transport/rcmgr_test.go:102]: I think this is implicit:
```suggestion
						streamScope.EXPECT().SetService(gomock.Any()).MaxTimes(1)
```

File: p2p/test/transport/rcmgr_test.go [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+func TestResourceManagerIsUsed(t *testing.T) {
+	for _, tc := range transportsToTest {
+		t.Run(tc.Name, func(t *testing.T) {
+			for _, testDialer := range []bool{true, false} {
+				t.Run(tc.Name+fmt.Sprintf(" test_dialer=%v", testDialer), func(t *testing.T) {
+
+					reservedMemory := atomic.Uint32{}
+					releasedMemory := atomic.Uint32{}
+					defer func() {
+						require.Equal(t, reservedMemory.Load(), releasedMemory.Load())
```
Comment (marten-seemann) [inline p2p/test/transport/rcmgr_test.go:28]: Should we also check that it's not 0?

File: p2p/test/transport/rcmgr_test.go (hunk 1 above)
Comment (marten-seemann) [reply p2p/test/transport/rcmgr_test.go:26]: Safer: Make it a signed int.

**Review by marten-seemann — APPROVED** [review]
This is fantastic! This flow is not immediately obvious, and it's great to have a test that makes sure that we're actually calling the right methods on the resource manager.

Just some super minor nits.

**Comment by MarcoPolo** [comment]
Possibly flaky test here, will investigate

**Review by marten-seemann — APPROVED** [review]

**Pull request merged**